common/queue/src/main/java/org/thingsboard/server/queue/discovery/TbApplicationEventListener.java

```diff
@@ -15,21 +15,27 @@
  */
 package org.thingsboard.server.queue.discovery;
 
-import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.context.ApplicationListener;
 import org.thingsboard.server.queue.discovery.event.TbApplicationEvent;
 
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
-@Slf4j
 public abstract class TbApplicationEventListener<T extends TbApplicationEvent> implements ApplicationListener<T> {
 
     private int lastProcessedSequenceNumber = Integer.MIN_VALUE;
     private final Lock seqNumberLock = new ReentrantLock();
 
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
     @Override
     public void onApplicationEvent(T event) {
+        if (!filterTbApplicationEvent(event)) {
```

Comment:
The point is to change `lastProcessedSequenceNumber` only for events with matching `serviceType`